src/engines/aurora/gui.h

```diff
@@ -63,6 +63,12 @@ class GUI : boost::noncopyable {
 	/** Process the current event queue. */
 	uint32 processEventQueue();
 
+	/** Add a widget.
+	 *
+	 *  @param show Show it immediately?
+	 */
+	void addWidget(Widget *widget, bool show = false);
```

Comment:
I don't have time today to look deeper into it, but is this really necessary? I don't like having methods that change the GUI itself public.